src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTreeUI.java

```diff
@@ -3280,6 +3280,9 @@ public Rectangle getNodeDimensions(Object value, int row,
                      expanded, treeModel.isLeaf(value), row,
                      false);
                 if(tree != null) {
+                    for (int i = 0; i < rendererPane.getComponentCount(); i++) {
+                        rendererPane.remove(i);
+                    }
```

Comment:
~~~java
for (int i = 0; i < rendererPane.getComponentCount(); i++) {
   rendererPane.remove(i);
}
~~~

This code will not work for two or more children!

For two children It does:
1. `i = 0`
2. `remove(i)` removes first child, which shifts other children and decrements component count to `1`!
3. `i++` so `i` is now `1`
4. check `i < rendererPane.getComponentCount()`, which is `1 < 1`. So loop is quit and one child is not removed.

For three children It does:
1. `i = 0`
2. `remove(i)` removes first child, which shifts other children and decrements component count to `2`!
3. `i++` so `i` is now `1`
4. check `i < rendererPane.getComponentCount()`, which is `1 < 2`
5. `remove(i)` removes second child, which was the third child initially and decrements component count to `1`
6. `i++` so `i` is now `2`
7. check `i < rendererPane.getComponentCount()`, which is `2 < 1` . So loop is quit and one child is not removed.